Automattic/wp-calypso
# Domain Management i1: Migrate DomainsTable requests from wpcom-proxy-request to lib/wp

Should fix issue reported p1695159601536029-slack-C04H4NY6STW

## Proposed Changes

* Switch from calling the wpcom proxy iframe directly to going through the `lib/wp` library.

This library has extra logic for deciding whether to use the wpcom proxy iframe or not. For example if Calypso is in oauth mode it doesn't use the proxy iframe but instead will send requests with an authorisation header.

`lib/wp` also has code for handling support sessions, which adds extra query params to requests.

We can't reference `lib/wp` directly in the `data-stores` and `domains-table` package because that imports Calypso code directly into reusable packages. When I tried it broke the HappyBlocks build (which must use the `data-stores` package) because it's build config mustn't know how to deal with building `lib/wp`.
Instead I've had to add fetching props to `<DomainsTable>` for each of the network calls we make. The Calypso app creates fetching functions using `wp` and passes them to the `<DomainsTable>` to use.

## Testing Instructions



You can test by putting Calypso into oauth mode using the dev menu:
![CleanShot 2023-09-20 at 11 00 06@2x](https://github.com/Automattic/wp-calypso/assets/1500769/ced40f5a-d956-468b-b3b1-2fb1efdaca43)

Test everything that makes a network request:
- Loading the page
- Scrolling the page
- Bulk update auto-renew
- Bulk update contact details
- Polling for bulk action to complete
- Clearing the bulk update success message

You should notice in the network tools that requests do not send cookies but send a `Bearer` token in the authorise header.

You can also test with a support session. See PCYsg-lsl-p2#usage. Use network admin to start a support session with a test user. After the incognito browser window has opened replace `https://wordpress.com` in the address bar with `http://calypso.localhost:3000`. You should still see the floating window indicating you're in a support session.

Run the same list of tests as above. You should notice in the network tools that requests do not send cookies but a special `X-Support-Session` token.

## Pre-merge Checklist
